website/api/recursion.md

```diff
@@ -0,0 +1,123 @@
+suc6-->j3
+
+j1-->suc11("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+j2-->suc12("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+j3-->suc13("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+
+suc11-->j11("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ApiClient.html#method.join'>.join()</a>")
+suc12-->j11
+
+j11-->suc21("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+
+suc21-->j21("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ApiClient.html#method.join'>.join()</a>")
+suc13-->j21
+
+j21-->suc31("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+
+suc31-->id("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ApiClient.html#method.identity_p254'>.identity_p254()</a>")
+
+id-->suc41("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+
+suc41-->compress("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ApiClient.html#method.compress'>.compress()</a>")
+
+compress-->groth16("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.Groth16Receipt.html'>Groth16Receipt</a>")
+```
+
+To summarize the diagram:
+
+1. The program is _executed_, resulting in a collection of _Segments_.
+1. Each _Segment_ is proven, resulting in a _SegmentReceipt_.
+1. Each _SegmentReceipt_ is _lifted_, resulting in a _SuccinctReceipt_.
+1. Pairs of _SuccinctReceipts_ are _joined_, resulting in another _SuccinctReceipt_. This continues until a single _SuccinctReceipt_ remains.
+1. The final _SuccinctReceipt_ is passed through _identity_p254_, which prepares for Groth16 proving.
+1. The _SuccinctReceipt_ is _compressed_, generating a _Groth16Receipt_.
+
+The _Groth16Receipt_ can now be posted on-chain and verified by the [RISC Zero Verifier Contract].
+
+## Recursive Circuit Architecture
+
+RISC Zero's zkVM consists of three circuits.
+
+1. The RISC-V Circuit is a STARK circuit that proves correct execution of RISC-V programs.
+1. The Recursion Circuit is a separate STARK circuit, that's designed to efficiently generate proofs for the verification of STARK proofs and to support the integration of custom accelerator circuits into the zkVM.
+   This circuit has a similar architecture to the RISC-V Circuit, but with fewer columns and a much simpler instruction set.
+   The Recursion Circuit supports a number of programs, including [lift], [join], [resolve], and [identity_p254].
+   The same [proof system] is used for both the RISC-V Circuit and the Recursion Circuit.
+1. The STARK-to-SNARK Circuit is an R1CS circuit that proves verification of proofs from the _identity_p254_ program.
```

Comment:
It's somewhat more accurate to say that it proves verification of proofs from the _recursion circuit_. It's execution is independent of the program that the recursion circuit ran, which is committed to by the control ID passed in as a public input. (This is good, because it means we can change the program without breaking the trusted setup)